src/utils/account.js

```diff
@@ -54,32 +54,56 @@ export const getActiveTokenAccount = state => ({
 export const truncateAddress = address =>
   address.replace(regex.lskAddressTrunk, '$1...$3');
 
-export const calculateLockedBalance = ({ votes = [] }) =>
-  votes.reduce((acc, vote) => acc + parseInt(vote.amount, 10), 0);
+/**
+ * calculates the balance locked in votes
+ */
+export const calculateBalanceLockedInVotes = (votes = {}) =>
+  Object.values(votes).reduce((total, vote) => (total + vote.confirmed), 0);
+
+/**
+ * calculates balance locked for the account in unvotes
+*/
+export const calculateBalanceLockedInUnvotes = (unlocking = []) =>
+  unlocking.reduce((acc, vote) => acc + parseInt(vote.amount, 10), 0);
 
 // TODO handle delegate punishment when Lisk Service is ready
 export const getDelayedAvailability = isSelfVote => (isSelfVote
   ? unlockTxDelayAvailability.selfUnvote : unlockTxDelayAvailability.unvote);
 
-export const isBlockHeightReached = ({ unvoteHeight, delegateAddress }, currentBlock, address) => {
+export const isBlockHeightReached = ({ height, delegateAddress }, currentBlock, address) => {
```

Comment:
If the `vote.height.end` is the block when the locked balance is available to be unlocked then we should just compare ```currentBlock.height <= vote.height.end```